components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SearchSuggestionProvider.kt

```diff
@@ -106,7 +113,7 @@ class SearchSuggestionProvider(
             chips.add(AwesomeBar.Suggestion.Chip(text))
         }
 
-        result?.forEach { title ->
+        result?.take(limit - chips.size)?. forEach { title ->
```

Comment:
Could be possible that `chips.size` is bigger than `limit`? If it possible will produce a negative number and cause an error.

Reply:
At this point `chips` either contains no items (0) or the default item (1) - the other items will be added in this loop. The only way it could be bigger is if someone chooses a limit that is 0 (and the default item is already added) or a negative number. Possible - but an error case nevertheless. I guess we could add a check that a limit is a positive number and at least 1. I'll file an issue.

Reply:
At this point `chips` either contains no items (0) or the default item (1) - the other items will be added in this loop. The only way it could be bigger is if someone chooses a limit that is 0 (and the default item is already added) or a negative number. Possible - but an error case nevertheless. I guess we could add a check that a limit is a positive number and at least 1. I'll file an issue.

Reply:
#2432

Reply:
👍